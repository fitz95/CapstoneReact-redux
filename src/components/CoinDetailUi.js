import React from 'react';
import PropTypes from 'prop-types';
import { BiRightArrowCircle } from 'react-icons/bi';
import '../css/coindetailui.css';

function CoinDetailUi({
  currentPrice,
  Alltimehigh,
  Alltimelow,
  fullydillutedvaluation,
  marketcap,
  pricechangepercentageyearly,
  totalvolume,
}) {
  return (
    <div className="coindetail">
      <div className="coindetaildiv">
        <h1 className="name">Current Price</h1>
        <div className="subcoindetaildiv">
          {currentPrice && (
          <h1 className="name">
            {currentPrice}
            $
          </h1>
          )}
          <BiRightArrowCircle className="icon" />
        </div>
      </div>
      <div className="coindetaildiv">
        <h1 className="name">All Time High</h1>
        <div className="subcoindetaildiv">
          {Alltimehigh && (
            <h1 className="name">{Alltimehigh}</h1>
          )}
          <BiRightArrowCircle className="icon" />
        </div>
      </div>
      <div className="coindetaildiv">
        <h1 className="name">All Time Low</h1>
        <div className="subcoindetaildiv">
          {Alltimelow && (
            <h1 className="name">{Alltimelow}</h1>
          )}
          <BiRightArrowCircle className="icon" />
        </div>
      </div>
      <div className="coindetaildiv">
        <h1 className="name">Fully Diluted Evaluation</h1>
        <div className="subcoindetaildiv">
          {fullydillutedvaluation && (
            <h1 className="name">{fullydillutedvaluation}</h1>
          )}
          <BiRightArrowCircle className="icon" />
        </div>
      </div>
      <div className="coindetaildiv">
        <h1 className="name">Market Cap</h1>
        <div className="subcoindetaildiv">
          {marketcap && (
            <h1 className="name">{marketcap}</h1>
          )}
          <BiRightArrowCircle className="icon" />
        </div>
      </div>
      <div className="coindetaildiv">
        <h1 className="name">Price Change Per Year</h1>
        <div className="subcoindetaildiv">
          {pricechangepercentageyearly && (
            <h1 className="name">{pricechangepercentageyearly}</h1>
          )}
          <BiRightArrowCircle className="icon" />
        </div>
      </div>
      <div className="coindetaildiv">
        <h1 className="name">Total Volume</h1>
        <div className="subcoindetaildiv">
          {totalvolume && (
            <h1 className="name">{totalvolume}</h1>
          )}
          <BiRightArrowCircle className="icon" />
        </div>
      </div>
    </div>
  );
}
CoinDetailUi.propTypes = {
  currentPrice: PropTypes.number.isRequired,
  Alltimehigh: PropTypes.number.isRequired,
  Alltimelow: PropTypes.number.isRequired,
  fullydillutedvaluation: PropTypes.number.isRequired,
  marketcap: PropTypes.number.isRequired,
  pricechangepercentageyearly: PropTypes.number.isRequired,
  totalvolume: PropTypes.number.isRequired,
};

export default CoinDetailUi;
