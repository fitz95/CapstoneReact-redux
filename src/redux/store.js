import { configureStore, combineReducers } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import coinsReducer from './coins/coinsSlice';
import coinDetailsReducer from './coindetails/coinDetailsSlice';

const store = configureStore({
  reducer: combineReducers({
    coins: coinsReducer,
    coinDetails: coinDetailsReducer,
  }),
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
