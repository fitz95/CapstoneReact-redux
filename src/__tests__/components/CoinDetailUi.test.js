import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from 'src/redux/store';
import CoinDetailUi from 'src/components/CoinDetailUi';

describe('RocketsLists', () => {
  it('renders correctly for reserved state', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <CoinDetailUi />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
