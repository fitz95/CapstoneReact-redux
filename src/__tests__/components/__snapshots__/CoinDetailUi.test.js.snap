// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RocketsLists renders correctly for reserved state 1`] = `
<div
  className="coindetail"
>
  <div
    className="coindetaildiv"
  >
    <h1
      className="name"
    >
      Cuurent Price
    </h1>
    <div
      className="subcoindetaildiv"
    >
      <h1
        className="name"
      >
        $
      </h1>
      <svg
        className="icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.999 1.993c-5.514.001-10 4.487-10 10.001s4.486 10 10.001 10c5.513 0 9.999-4.486 10-10 0-5.514-4.486-10-10.001-10.001zM12 19.994c-4.412 0-8.001-3.589-8.001-8s3.589-8 8-8.001C16.411 3.994 20 7.583 20 11.994c-.001 4.411-3.59 8-8 8z"
        />
        <path
          d="M12 10.994H8v2h4V16l4.005-4.005L12 7.991z"
        />
      </svg>
    </div>
  </div>
  <div
    className="coindetaildiv"
  >
    <h1
      className="name"
    >
      All Time High
    </h1>
    <div
      className="subcoindetaildiv"
    >
      <h1
        className="name"
      />
      <svg
        className="icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.999 1.993c-5.514.001-10 4.487-10 10.001s4.486 10 10.001 10c5.513 0 9.999-4.486 10-10 0-5.514-4.486-10-10.001-10.001zM12 19.994c-4.412 0-8.001-3.589-8.001-8s3.589-8 8-8.001C16.411 3.994 20 7.583 20 11.994c-.001 4.411-3.59 8-8 8z"
        />
        <path
          d="M12 10.994H8v2h4V16l4.005-4.005L12 7.991z"
        />
      </svg>
    </div>
  </div>
  <div
    className="coindetaildiv"
  >
    <h1
      className="name"
    >
      All Time Low
    </h1>
    <div
      className="subcoindetaildiv"
    >
      <h1
        className="name"
      />
      <svg
        className="icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.999 1.993c-5.514.001-10 4.487-10 10.001s4.486 10 10.001 10c5.513 0 9.999-4.486 10-10 0-5.514-4.486-10-10.001-10.001zM12 19.994c-4.412 0-8.001-3.589-8.001-8s3.589-8 8-8.001C16.411 3.994 20 7.583 20 11.994c-.001 4.411-3.59 8-8 8z"
        />
        <path
          d="M12 10.994H8v2h4V16l4.005-4.005L12 7.991z"
        />
      </svg>
    </div>
  </div>
  <div
    className="coindetaildiv"
  >
    <h1
      className="name"
    >
      All Time Low
    </h1>
    <div
      className="subcoindetaildiv"
    >
      <h1
        className="name"
      />
      <svg
        className="icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.999 1.993c-5.514.001-10 4.487-10 10.001s4.486 10 10.001 10c5.513 0 9.999-4.486 10-10 0-5.514-4.486-10-10.001-10.001zM12 19.994c-4.412 0-8.001-3.589-8.001-8s3.589-8 8-8.001C16.411 3.994 20 7.583 20 11.994c-.001 4.411-3.59 8-8 8z"
        />
        <path
          d="M12 10.994H8v2h4V16l4.005-4.005L12 7.991z"
        />
      </svg>
    </div>
  </div>
  <div
    className="coindetaildiv"
  >
    <h1
      className="name"
    >
      Market Cap
    </h1>
    <div
      className="subcoindetaildiv"
    >
      <h1
        className="name"
      />
      <svg
        className="icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.999 1.993c-5.514.001-10 4.487-10 10.001s4.486 10 10.001 10c5.513 0 9.999-4.486 10-10 0-5.514-4.486-10-10.001-10.001zM12 19.994c-4.412 0-8.001-3.589-8.001-8s3.589-8 8-8.001C16.411 3.994 20 7.583 20 11.994c-.001 4.411-3.59 8-8 8z"
        />
        <path
          d="M12 10.994H8v2h4V16l4.005-4.005L12 7.991z"
        />
      </svg>
    </div>
  </div>
  <div
    className="coindetaildiv"
  >
    <h1
      className="name"
    >
      Price Change Per Year
    </h1>
    <div
      className="subcoindetaildiv"
    >
      <h1
        className="name"
      />
      <svg
        className="icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.999 1.993c-5.514.001-10 4.487-10 10.001s4.486 10 10.001 10c5.513 0 9.999-4.486 10-10 0-5.514-4.486-10-10.001-10.001zM12 19.994c-4.412 0-8.001-3.589-8.001-8s3.589-8 8-8.001C16.411 3.994 20 7.583 20 11.994c-.001 4.411-3.59 8-8 8z"
        />
        <path
          d="M12 10.994H8v2h4V16l4.005-4.005L12 7.991z"
        />
      </svg>
    </div>
  </div>
  <div
    className="coindetaildiv"
  >
    <h1
      className="name"
    >
      Total Volume
    </h1>
    <div
      className="subcoindetaildiv"
    >
      <h1
        className="name"
      />
      <svg
        className="icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.999 1.993c-5.514.001-10 4.487-10 10.001s4.486 10 10.001 10c5.513 0 9.999-4.486 10-10 0-5.514-4.486-10-10.001-10.001zM12 19.994c-4.412 0-8.001-3.589-8.001-8s3.589-8 8-8.001C16.411 3.994 20 7.583 20 11.994c-.001 4.411-3.59 8-8 8z"
        />
        <path
          d="M12 10.994H8v2h4V16l4.005-4.005L12 7.991z"
        />
      </svg>
    </div>
  </div>
</div>
`;
